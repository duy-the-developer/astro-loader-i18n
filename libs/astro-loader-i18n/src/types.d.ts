export type NormalizeLocales<T extends Locales> = {
  [K in keyof T]: T[K] extends string ? T[K] : T[K] extends {
      codes: Array<string>;
  } ? T[K]['codes'][number] : never;
}[number];
