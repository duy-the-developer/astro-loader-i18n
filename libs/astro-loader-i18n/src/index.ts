export { i18nLoader } from "./loaders/i18n-loader";
export { i18nInfileSchema } from "./schemas/i18n-infile-schema";
export { i18nLoaderSchema } from "./schemas/i18n-loader-schema";
export { localePaths } from "./paths/locale-paths";
export { translationPaths } from "./paths/translation-paths";
export { mergeGetStaticPathsResults } from "./utils/merge";
export { createCollection } from "./collections/create-collection";
