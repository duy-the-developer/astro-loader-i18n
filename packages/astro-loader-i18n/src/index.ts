export { i18nFolderLoader } from "./loaders/i18n-folder-loader";
export { i18nSchema } from "./schemas/i18n-schema";
