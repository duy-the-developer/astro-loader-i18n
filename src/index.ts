export { i18nFolderLoader } from "./loaders/i18n-folder-loader";
