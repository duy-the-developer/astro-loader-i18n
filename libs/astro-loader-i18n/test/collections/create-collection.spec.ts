import { describe, expect, it } from "vitest";
import { createCollection } from "../../src";

describe("createCollection", () => {
  it("should create a collection based on the provided locales and route pattern", () => {
    const locales = ["en", "de"];
    const routePattern = "/[...locale]/blog/[slug]";
    expect(createCollection({ locales, routePattern })).toMatchSnapshot();
  });
});
