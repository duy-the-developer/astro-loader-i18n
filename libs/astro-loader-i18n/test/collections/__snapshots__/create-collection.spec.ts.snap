// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createCollection > should create a collection based on the provided locales and route pattern 1`] = `
[
  {
    "locale": "en",
    "translationId": "/[...locale]/blog/[slug]",
  },
  {
    "locale": "de",
    "translationId": "/[...locale]/blog/[slug]",
  },
]
`;
